src/WooCommerce.php

```diff
@@ -1513,4 +1514,14 @@ private static function getRepricingOptionsDropdownValues(\WC_Product $product)
     return $dropdown_values;
   }
 
+    /**
+   * Invalidate the object cache in Woocommerce order status manager
+   * 
+   * @implements save_post_wc_order_status
+   * 
+   */
+  public static function save_post_wc_order_status() {
+      wp_cache_delete( 'wc_order_status_manager_order_status_posts');
```

Comment:
I think we should check like you did before if we have someone in the cache first, and if we do we can delete it.